etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestrator.java

```diff
@@ -15,45 +23,89 @@ public class PartnerMetadataOrchestrator {
     private static final PartnerMetadataOrchestrator INSTANCE = new PartnerMetadataOrchestrator();
 
     @Inject PartnerMetadataStorage partnerMetadataStorage;
+    @Inject ReportStreamEndpointClient rsclient;
+    @Inject Formatter formatter;
 
     public static PartnerMetadataOrchestrator getInstance() {
         return INSTANCE;
     }
 
     private PartnerMetadataOrchestrator() {}
 
-    public void updateMetadataForReceivedOrder(String submissionId, Order<?> order)
+    public void updateMetadataForReceivedOrder(String receivedSubmissionId, Order<?> order)
             throws PartnerMetadataException {
-        // will call the RS history API given the submissionId (albeit, right now this won't work
-        // given the way RS works).
-        // from the history API response, extract the sender (organization + sender client), and
-        // time received.
+        // will call the RS history API given the submissionId, currently blocked by:
+        // https://github.com/CDCgov/prime-reportstream/issues/12624
+        // from the response, extract the "sender" and "timestamp" (timeReceived)
         // we will calculate the hash.
         // then we call the metadata storage to save this stuff.
 
+        String sender = "unknown";
+        Instant timeReceived = Instant.now();
+        String hash = String.valueOf(order.hashCode());
```

Comment:
This is not a secure method of hashing the order which is sensitive info.  We should figure out the type of hash we want to use and implement something like the below link based on the algorithm. 
[hashing](https://reflectoring.io/creating-hashes-in-java/)

Reply:
Good point, though I think this hash will be more of a placeholder for now. RS should be the one creating a sending the hash to us, since they'll have the original order